Unexpected invocation of UDF

**Describe the bug**

```
Caused by: java.lang.RuntimeException: poll record batch error: Execution error: native execution panics: Execution error: Execution error: output_with_sender[Project] error: Execution error: output_with_sender[Project]: output() returns error: External error: Java exception thrown at native-engine/datafusion-ext-exprs/src/spark_udf_wrapper.rs:216: java.lang.RuntimeException: Error loading expression '/data/k1'
	at org.apache.auron.jni.JniBridge.nextBatch(Native Method)
	at org.apache.auron.jni.AuronCallNativeWrapper.loadNextBatch(AuronCallNativeWrapper.java:118)
	... 19 more
```

**To Reproduce**
```sql
create table tmp_xml ( id int ,c1 string) stored as parquet ;

insert overwrite table tmp_xml values (1,'a'),(5,'<'),(5,'<'),(5,'<'),(5,'<'),(5,'<'),(5,'<'),(2,'<data><k1>v1</k1></data>'),(3,'<daa>'),(4,'>'),(5,'<'),(5,'<'),(5,'<'),(5,'<'),(5,'<'),(5,'<');
```

```sql
select * , case when id=2 and xpath_string(c1,'/data/k1')= 'v1' then 'xml' else 'no_xml' end as extracted_value from tmp_xml;
```



**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

**Screenshots**
<!--
If applicable, add screenshots to help explain your problem.
-->

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/spark-extension/src/main/java/org/apache/auron/spark/configuration/SparkAuronConfiguration.java
+++ b/spark-extension/src/main/java/org/apache/auron/spark/configuration/SparkAuronConfiguration.java
@@ -238,6 +238,13 @@ public class SparkAuronConfiguration extends AuronConfiguration {
             .booleanType()
             .defaultValue(false);
 
+    public static final ConfigOption<Boolean> FORCE_SHORT_CIRCUIT_AND_OR = ConfigOptions.key(
+                    "auron.forceShortCircuitAndOr")
+            .description("force using short-circuit evaluation (PhysicalSCAndExprNode/PhysicalSCOrExprNode) "
+                    + "for And/Or expressions, regardless of whether rhs contains HiveUDF. ")
+            .booleanType()
+            .defaultValue(false);
+
     private final SparkConf sparkConf;
 
     public SparkAuronConfiguration(SparkConf conf) {

--- a/spark-extension/src/main/scala/org/apache/spark/sql/auron/NativeConverters.scala
+++ b/spark-extension/src/main/scala/org/apache/spark/sql/auron/NativeConverters.scala
@@ -769,7 +769,10 @@ object NativeConverters extends Logging {
         }
 
       // if rhs is complex in and/or operators, use short-circuiting implementation
-      case And(lhs, rhs) if rhs.find(HiveUDFUtil.isHiveUDF).isDefined =>
+      // or if forceShortCircuitAndOr is enabled, always use short-circuiting
+      case And(lhs, rhs)
+          if sparkAuronConfig.getBoolean(SparkAuronConfiguration.FORCE_SHORT_CIRCUIT_AND_OR)
+            || rhs.find(HiveUDFUtil.isHiveUDF).isDefined =>
         buildExprNode {
           _.setScAndExpr(
             pb.PhysicalSCAndExprNode
@@ -777,7 +780,9 @@ object NativeConverters extends Logging {
               .setLeft(convertExprWithFallback(lhs, isPruningExpr, fallback))
               .setRight(convertExprWithFallback(rhs, isPruningExpr, fallback)))
         }
-      case Or(lhs, rhs) if rhs.find(HiveUDFUtil.isHiveUDF).isDefined =>
+      case Or(lhs, rhs)
+          if sparkAuronConfig.getBoolean(SparkAuronConfiguration.FORCE_SHORT_CIRCUIT_AND_OR)
+            || rhs.find(HiveUDFUtil.isHiveUDF).isDefined =>
         buildExprNode {
           _.setScOrExpr(
             pb.PhysicalSCOrExprNode

